source/run-command.txt

```diff
@@ -0,0 +1,182 @@
+   
+   To perform administrative tasks, use the :mongosh:`MongoDB Shell </>` instead
+   of the {+driver-short+}. Calling the ``db.runCommand()`` method inside the
+   shell is the preferred method to issue database commands, as it provides a
+   consistent interface between the shell and drivers.
+
+Execute a Command
+-----------------
+
+To run a database command, you must specify the command and any relevant
+parameters in a BSON document, then pass this BSON document to a command
+execution method. The {+driver-short+} provides the following methods to run
+database commands:
+
+- ``mongoc_client_command_simple()``, which returns the first document from the
+  result cursor into ``reply``. This method is a simplified interface that sends
+  a raw command to the server.
+- ``mongoc_client_command_with_opts()``, which runs a command on the server and
+  interprets ``opts`` according to this MongoDB server version. This method
+  offers flexibiilty by allowing additional options.
+
+The following code shows how you can use the ``mongoc_client_command_simple()``
+method to run the ``hello`` command, which returns information about the current
+member's role in the replica set, on a database:
+
+.. literalinclude:: /includes/usage-examples/run-command.c
+    :language: c
+    :dedent:
+    :start-after: start-execute-command
+    :end-before: end-execute-command
+
+For a full list of database commands and corresponding parameters, see the
+:ref:`Additional Information <c-addl-info-runcommand>` section.
+
+Command Options
+---------------
+
+You can specify optional command behavior for the
+``mongoc_client_command_with_opts()`` method.
+
+The ``mongoc_client_command_with_opts()`` method accepts a BSON document for the
+``opts`` parameter. This can include command options such as read concerns,
+write concerns, and session options. 
+
+You can pass a BSON document with the following options:
+
+- ``readConcern``
+- ``writeConcern``
+- ``sessionId``
+- ``collation``
+- ``serverId``
+
+To learn more about a command and the options that it accepts, locate the
+command and follow the link in the :manual:`Database Commands
+</reference/command/>` section of the Server manual.
+
+The following code shows you how to specify a ``grantRolesToUsers`` command with a
+``majority`` write concern:
+
+.. literalinclude:: /includes/usage-examples/run-command.c
+    :language: c
+    :dedent:
+    :start-after: start-execute-command-with-options
+    :end-before: end-execute-command-with-options
+
+.. note:: Read Preference
+
+   The ``mongoc_client_command_simple()`` and
+   ``mongoc_client_command_with_opts()`` methods ignore the read preference
+   setting you may have set on your ``Db`` object. By default, these methods use
+   the ``primary`` read preference.
+
+   To specify a read preference other than the primary read preference, you need
+   to explicity pass it as an argument. The following code demonstrates how to
```

Comment:
Style Guide - use "must" instead of "need to": https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#std-term-must
```suggestion
   To specify a read preference other than the primary read preference, you must explicity pass it as an argument. The following code demonstrates how to
```